src/core/profiling/endoscope_backend.c

```diff
@@ -0,0 +1,577 @@
+#include "src/core/profiling/endoscope_backend.h"
+
+#include <grpc/support/time.h>
+#include <stdio.h>
+#include <string.h>
+#include <sys/time.h>
+#include <sys/syscall.h>
+#include <unistd.h>
+
+static gpr_int64 grpc_endo_invalid64 = 0;  /* value written here will be ignored */
+
+static const char *grpc_endo_warning_str = "MARKER Task Atom Thread ";
+typedef enum grpc_endo_warning_enum {  /* values match positions */
```

Comment:
No, that's my own simple implementation of error message, since it's very efficient to write and to run, although ugly.

It works like this:
str = "A_BB_CCC_";
msg = "______";
enum {A = 0; B = 2; C = 5;};
Given error code B, replace msg[i] with str[i] start from i = B until reach "_". Then we get "__BB_____" as warning message.